SIPS/sip-26.md

```diff
@@ -0,0 +1,194 @@
+---
+sip: 26
+title: Non-EVM protocol support
+status: Draft
+author: Daniel Rocha (@danroc), Frederik Bolding (@FrederikBolding), Alex Donesky (@adonesky1)
+created: 2024-08-28
+---
+
+## Abstract
+
+This SIP presents an architecture to enable Snaps to expose blockchain-specific
+methods to dapps, extending MetaMask's functionality to support a multichain
+ecosystem.
+
+## Motivation
+
+Currently, MetaMask is limited to EVM-compatible networks. This proposal aims
+to empower developers, both first- and third-party, to use Snaps to add native
+support for non-EVM-compatible chains within MetaMask.
+
+## Specification
+
+> Formal specifications are written in TypeScript.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### High-Level architecture
+
+The diagram below represents a high-level architecture of how the RPC Router
+integrates inside MetaMask to allow Snaps to expose protocol methods to dapps
+and the MetaMask clients.
+
+![High-level architecture](../assets/sip-26/components-diagram.png)
+
+- **Account Snaps**: Snaps that implement the [Keyring API][keyring-api] and are responsible
+  for signing requests and managing accounts.
+
+- **Protocol Snaps**: Snaps that implement protocol methods that do not require
+  an account to be executed.
+
+- **RPC Router**: Native component that forwards RPC requests to the
+  appropriate Protocol Snap or Account Snap.
+
+- **Keyring Controller**: Native component responsible for forwarding signing
+  requests to the appropriate keyring implementation.
+
+- **Accounts Controller**: Native component responsible for managing accounts
+  inside MetaMask. It stores all non-sensitive account information.
+
+- **Snaps Keyring**: Native component that acts as a bridge between the
+  Keyring Controller and the Account Snaps.
+
+### Components
+
+Here is a brief description of the components involved in this architecture
+which will require to be implemented or modified.
+
+#### RPC Router
+
+The RPC Router will be a new native component responsible for routing JSON-RPC
+requests to the appropriate Snap or keyring.
```

Comment:
Is it confusing to separate `keyring` from Snap here since all the keyrings we route to are Snap keyrings?
```suggestion
requests to the appropriate Snap.
```